### ISSUE TITLE
Usage, how to catch errors

### ISSUE THREAD

**Issue by mariusk** [Q42/RxPromise#7]
I'm having some trouble figuring out how this library is supposed to be used. Can anybody with a clue provide a canonical example of Promise.async should be used, with a proper success and failure handler, including how the error would be signaled (calling `error` or `throwing`?)? I've tried calling `then` with two callback functions, and calling `onError` on the return value of `Promise.async`, but without much success in handling errors. TIA.

**Comment by tsuijten** [comment]
I'll give you some examples, hopefully they make sense to you.

#### Java 7 Example that has a success and error handler that always succeeds

``` java
Promise.async(new Callable<String>() {
    @Override
    public String call() throws Exception {
        // Long running operation
        return "result";
    }
}).onSuccess(new Action1<String>() {
    @Override
    public void call(String s) {
        // Prints "result"
        System.out.println(s);
    }
}).onError(new Action1<Throwable>() {
    @Override
    public void call(Throwable throwable) {
        // Prints the error
        throwable.printStackTrace();
    }
});
```

#### Java 7 Example that has a success and error handler that always fails

``` java
Promise.async(new Callable<String>() {
    @Override
    public String call() throws Exception {
        // Test with an error
        throw new RuntimeException("oops");
    }
}).onSuccess(new Action1<String>() {
    @Override
    public void call(String s) {
        // Prints "result"
        System.out.println(s);
    }
}).onError(new Action1<Throwable>() {
    @Override
    public void call(Throwable throwable) {
        // Prints the error
        throwable.printStackTrace();
    }
});
```

#### First example rewritten in Java 8 with lambda's

``` java
Promise.async(() -> "result")
        .onSuccess(System.out::println)
        .onError(Throwable::printStackTrace);
```

_could also be written as_

``` java
Promise.async(() -> "result")
        .then(System.out::println, Throwable::printStackTrace);
```

Please let me know if these examples help you to understand the library

**Comment by mariusk** [comment]
There's definitively something going on at my end, because I when I tried the two callable form of `then`, the error handler did not get called when an exception got called. For the heck of it, I also tried adding an `.onError` handler (but after `.then` which possibly doesn't even make any sense), but no success there either. It could be something related to me trying to do this using Kotlin (not Java) as well. But based on your comment it seems I've understood how it's supposed to work, even though I haven't been able to get it working. In the meantime I've worked around it using another Promise library (Kotlin native, Kovenant) and I'm seeing no trouble there. I'll see if I can give this one another try.
